util/util.go

```diff
@@ -0,0 +1,60 @@
+package util
```

Comment:
Could we change the package name to something different? `util` is very generic.

Reply:
I think it was me who introduced a `util` when moving predicates to its own package in https://github.com/rancher-sandbox/rancher-turtles/pull/50/commits/6c01783876743422e93be6e8e4cf675a6f4ee708 following on how CAPI does it: https://github.com/kubernetes-sigs/cluster-api/tree/main/util